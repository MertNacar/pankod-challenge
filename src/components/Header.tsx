import React from "react";
import { withRouter } from "react-router";

const Header: React.FC<any> = (props) => {
  function goHomeScreen() {
    props.history.goBack();
  }
  return (
    <header>
      <nav id="myHeader" className="shadow navbar navbar-dark bg-primary">
        <button
          className="px-5 text-light btn btn-outline-primary"
          onClick={() => goHomeScreen()}
        >
          <h4>Demo Streaming</h4>
        </button>
        <span className="px-2">
          <p className="px-3 navbar-text text-light">Log in</p>

          <button className="px-2 btn btn-dark" type="submit">
            Start your free trial
          </button>
        </span>
      </nav>
    </header>
  );
};

export default withRouter(Header);
