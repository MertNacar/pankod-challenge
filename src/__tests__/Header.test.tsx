import React from "react";
import { shallow } from "enzyme";
import Header from "../components/Header";

describe("Header Component", () => {
  const wrapper = shallow(<Header />);

  it("is exists", () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  it("return Header", () => {
    expect(wrapper).toMatchSnapshot("Header");
  });

});
