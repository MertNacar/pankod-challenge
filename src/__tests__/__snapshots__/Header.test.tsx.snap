// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component return Header: Header 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
